### ISSUE TITLE
Amplitude & Angular - low performance

### ISSUE THREAD

**Issue by grzegorzKniazuk** [amplitude/Amplitude-TypeScript#899]
## Expected Behavior
The SDK shouldn't affect the overall performance of the application.

## Current Behavior
The performance of our application dropped significantly after adding the Amplitude SDK. For example: just checking a checkbox takes a few seconds on mobile devices. Even the screen scrolling wasn't smooth. We have removed SDK and everything works smoothly now.

## Steps to Reproduce
I haven't tried to reproduce this on another Angular project, so for now I'm adding a list of the dependencies we use in the project and class code what we use to integrate Amplitude. We import AmplitudeIntegrationModule directly into AppModule.  We call `init` method on application start using the initializer.

<img width="410" alt="asset1" src="https://github.com/user-attachments/assets/776a6fa7-2c4e-49b3-848d-7f23a5429b10">
<img width="600" alt="asset2" src="https://github.com/user-attachments/assets/fe85058b-0c31-4d55-a54a-1dbddfb107aa">

## Environment
- JS SDK Version: 2.11.7
- Installation Method: NPM
- Browser and Version: Chrome 130

**Comment by Mercy811** [comment]
Hi @grzegorzKniazuk, does the same issue happens when you open the page on a laptop browser? Could you try to turn element interactions off by `init(apiKey, {autocapture. elementInteractions: false })` and turn autocapture off by ``init(apiKey, {autocapture: false })` and try again to see if reducing tracking events help?

**Comment by grzegorzKniazuk** [comment]
No, I think there is no issue on laptop or high-end mobile devices. I have tested app on a few devices. For example: on Samsung S23+ I did not see any performance issues, but I saw it on Samsung A52 (I use mobile Chrome browser).

I have turned off `elementInteractions` and everything works fine now. Looks like this is the source of the performance drop.
